src/Domain/DomainHelpers.hpp

```diff
@@ -0,0 +1,40 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+
+#pragma once
+
+#include <algorithm>
```

Comment:
All the includes need to be cleaned up.  Many of them should be in the cpp and some things that are needed are missing.